bigtable/tests/mutation_integration_test.cc

```diff
@@ -0,0 +1,508 @@
+
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key = "DelColKey";
+  std::vector<bigtable::Cell> created_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id1", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id1", 2000, "v-c-0-1", {}},
+      {row_key, column_family2, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c0-0-0", {}},
+      {row_key, column_family1, "column_id3", 3000, "v-c1-0-2", {}},
+  };
+
+  std::vector<bigtable::Cell> expected_cells{
+      {row_key, column_family1, "column_id1", 2000, "v-c-0-1", {}},
+      {row_key, column_family2, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c0-0-0", {}},
+      {row_key, column_family1, "column_id3", 3000, "v-c1-0-2", {}},
+  };
+
+  // Create records
+  CreateCells(*table, created_cells);
+
+  // end_time is not inclusive, only records with timestamp < time_end will be
+  // deleted
+  std::int64_t time_end = 2000;
+  // Delete the columns with column identifier as created_cells.at(1) (i.e.
+  // column_id1)
+  DeleteFromColumnEndingAt(*table, created_cells.at(1), time_end);
+
+  auto actual_cells = ReadRows(*table, bigtable::Filter::PassAllFilter());
+
+  DeleteTable(table_name);
+
+  CheckEqualUnordered(expected_cells, actual_cells);
+}
+
+TEST_F(MutationIntegrationTest, DeleteFromFamilyTest) {
+  std::string const table_name = "table4delfamily";
+
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key = "DelColKey";
+  std::vector<bigtable::Cell> created_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id1", 1000, "v-c-0-1", {}},
+      {row_key, column_family2, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c0-0-0", {}},
+      {row_key, column_family1, "column_id3", 3000, "v-c1-0-2", {}},
+  };
+
+  std::vector<bigtable::Cell> expected_cells{
+      {row_key, column_family2, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c0-0-0", {}},
+  };
+
+  // Create records
+  CreateCells(*table, created_cells);
+
+  // Delete all the records for family
+  DeleteFromFamily(*table, row_key, column_family1);
+
+  auto actual_cells = ReadRows(*table, bigtable::Filter::PassAllFilter());
+
+  DeleteTable(table_name);
+
+  CheckEqualUnordered(expected_cells, actual_cells);
+}
+
+TEST_F(MutationIntegrationTest, DeleteFromRowTest) {
+  std::string const table_name = "table4delrow";
+
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key1 = "DelRowKey1", row_key2 = "DelRowKey2";
```

Comment:
please move `row_key2` to a separate declaration, as in `std::string row_key2 = ...;`.